yt_dlp/extractor/teamcoco.py

```diff
@@ -1,57 +1,108 @@
 import json
+import re
 
 from .turner import TurnerBaseIE
 from ..utils import (
-    determine_ext,
     ExtractorError,
-    int_or_none,
+    clean_html,
+    determine_ext,
+    make_archive_id,
+    merge_dicts,
     mimetype2ext,
     parse_duration,
-    parse_iso8601,
-    qualities,
+    traverse_obj,
+    unified_timestamp,
+    urljoin,
+    url_or_none,
 )
 
 
-class TeamcocoIE(TurnerBaseIE):
-    _VALID_URL = r'https?://(?:\w+\.)?teamcoco\.com/(?P<id>([^/]+/)*[^/?#]+)'
+class TeamcocoBaseIE(TurnerBaseIE):
+    _QUALITIES = {
+        'low': (480, 272),
+        'sd': (640, 360),
+        'hd': (1280, 720),
+        'uhd': (1920, 1080),
+    }
+
+    def _get_formats_and_subtitles(self, info, video_id):
+        formats, subtitles = [], {}
+
+        for src in traverse_obj(info, ('src', ...), expected_type=dict):
```

Comment:
FYI:
```suggestion
        for src in traverse_obj(info, ('src', ..., {dict})):
```
is the recommended way now